src/eve/visitors.py

```diff
@@ -102,6 +109,16 @@ def apply(cls, tree, init_var, foo, bar=5, **kwargs):
 
     """
 
+    contexts: ClassVar[Optional[Tuple[ContextCallable, ...]]] = None
+
+    def _do_visit(self, node: concepts.TreeNode, method, **kwargs: Any) -> Any:
+        with contextlib.ExitStack() as stack:
+            for ctx in self.contexts or []:
+                stack.enter_context(ctx(self, node, kwargs))
+
+            return_value = method(node, **kwargs)
+        return return_value
```

Comment:
This should be equivalent:
```suggestion
            return method(node, **kwargs)
```